src/main/java/gregtech/common/blocks/MetaBlocks.java

```diff
@@ -121,11 +124,16 @@ private MetaBlocks() {}
 
     public static BlockBrittleCharcoal BRITTLE_CHARCOAL;
 
-    public static final Map<Material, BlockCompressed> COMPRESSED = new HashMap<>();
-    public static final Map<Material, BlockFrame> FRAMES = new HashMap<>();
-    public static final Collection<BlockOre> ORES = new ReferenceArrayList<>();
-    public static final Map<Material, BlockSurfaceRock> SURFACE_ROCK = new HashMap<>();
-    public static final Collection<BlockFluidBase> FLUID_BLOCKS = new ReferenceArrayList<>();
+    public static final Map<Material, BlockCompressed> COMPRESSED = new Object2ObjectOpenHashMap<>();
+    public static final Map<Material, BlockFrame> FRAMES = new Object2ObjectOpenHashMap<>();
+    public static final Map<Material, BlockSurfaceRock> SURFACE_ROCK = new Object2ObjectOpenHashMap<>();
+
+    public static final List<BlockCompressed> COMPRESSED_BLOCKS = new ArrayList<>();
+    public static final List<BlockFrame> FRAME_BLOCKS = new ArrayList<>();
+    public static final List<BlockSurfaceRock> SURFACE_ROCK_BLOCKS = new ArrayList<>();
```

Comment:
Are these lists needed? Why was the approach using `Map#values` for registering these no longer viable?

Reply:
The maps contain multiple instance of same block, and using raw `Map#values` will make whatever logic run multiple times. Previous approach used `distinct` stream operation to prevent the scenario, which is undesirable.